packages/neuron-wallet/src/utils/export-history.ts

```diff
@@ -0,0 +1,149 @@
+import fs from 'fs'
+import sqlite3 from 'sqlite3'
+import { get as getDescription } from 'database/leveldb/transaction-description'
+import i18n from 'locales/i18n'
+import shannonToCKB from 'utils/shannonToCKB'
+
+
+interface ExportHistoryParms {
+  walletID: string
+  dbPath: string
+  lockHashList?: string[]
+  filePath: string
+}
+const exportHistory = ({
+  walletID,
+  dbPath,
+  lockHashList = [],
+  filePath
+}: ExportHistoryParms): Promise<number> => {
+  return new Promise((resolve, reject) => {
+    if (!dbPath) {
+      return reject(new Error(`Database is not required`))
+    }
+
+    if (!Array.isArray(lockHashList)) {
+      return reject(new Error(`Lock hash list is expected to be an array`))
+    }
+
+    if (!filePath) {
+      return reject(new Error(`File Path is required`))
+    }
+
+    if (fs.existsSync(filePath)) {
+      fs.unlinkSync(filePath)
+    }
+
+    let total: number | undefined
+    let inserted = 0
+
+    const writeStream = fs.createWriteStream(filePath)
+    writeStream.write(
+      `${['time', 'block-number', 'tx-hash', 'tx-type', 'amount', 'description']
+        .map(label => i18n.t(`export-transactions.column.${label}`))}\n`
+    )
+
+    const db = new sqlite3.Database(dbPath, sqlite3.OPEN_READONLY)
```

Comment:
Do we have to sqlite3 directly here? Would it be better to use typeorm since we use typeorm in most of the cases?

Reply:
> Do we have to sqlite3 directly here? Would it be better to use typeorm since we use typeorm in most of the cases?

`Typeorm`'s `stream` doesn't work with `sqlite3` so I've to use `sqlite3#each` in case of bunches of data.